packages/store/addon/-private/ts-interfaces/schema-definition-service.ts

```diff
@@ -0,0 +1,14 @@
+/**
+  @module @ember-data/store
+*/
+import { RecordIdentifier } from './identifier';
+import { AttributesSchema, RelationshipsSchema } from './record-data-schemas';
+// we import the class not the interface because we expect
+// because we expect to use this internally with the more complete set
+// of APIs
+
+export interface SchemaDefinitionService {
+  doesTypeExist(modelName: string): boolean;
+  attributesDefinitionFor(identifier: RecordIdentifier | string): AttributesSchema;
```

Comment:
minor nit, overloads would allow separate variable name and documentation for the modelName case